test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NetCorePackageReferenceProjectTests.cs

```diff
@@ -0,0 +1,483 @@
+
+                // Act
+                var command = new RestoreCommand(request);
+                var result = await command.ExecuteAsync();
+                var packages = await project.GetInstalledPackagesAsync(CancellationToken.None);
+
+                // Asert
+                var exists = packages.Where(a => a.PackageIdentity.Equals(new PackageIdentity("packageA", new NuGetVersion("2.0.0"))));
+                Assert.True(exists.Count() == 1);
+            }
+        }
+
+        // No packages
+        [Fact]
+        public async Task GetInstalledVersion_WithoutPackages_ReturnsEmpty()
+        {
+            using (var testDirectory = TestDirectory.Create())
+            {
+                // Setup
+                var projectName = "project1";
+                var projectFullPath = Path.Combine(testDirectory.Path, projectName + ".csproj");
+
+                // Project
+                var projectCache = new ProjectSystemCache();
+                IVsProjectAdapter projectAdapter = (new Mock<IVsProjectAdapter>()).Object;
+                var project = CreateNetCorePackageReferenceProject(projectName, projectFullPath, projectCache);
+
+                var projectNames = GetTestProjectNames(projectFullPath, projectName);
+                var packageSpec = GetPackageSpecNoPackages(projectName, projectFullPath);
+                
+                // Restore info
+                var projectRestoreInfo = ProjectJsonTestHelpers.GetDGSpecFromPackageSpecs(packageSpec);
+                projectCache.AddProjectRestoreInfo(projectNames, projectRestoreInfo, new List<IAssetsLogMessage>());
+                projectCache.AddProject(projectNames, projectAdapter, project).Should().BeTrue();
+
+                // Package directories
+                var sources = new List<PackageSource>();
+                var packagesDir = new DirectoryInfo(Path.Combine(testDirectory, "globalPackages"));
+                var packageSource = new DirectoryInfo(Path.Combine(testDirectory, "packageSource"));
+                packagesDir.Create();
+                packageSource.Create();
+                sources.Add(new PackageSource(packageSource.FullName));
+
+                var logger = new TestLogger();
+                var request = new TestRestoreRequest(packageSpec, sources, packagesDir.FullName, logger);
+
+                // Act
+                var command = new RestoreCommand(request);
+                var result = await command.ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
+                var packages = await project.GetInstalledPackagesAsync(CancellationToken.None);
+
+                // Asert
+                packages.Should().BeEmpty();
+            }
+        }
+
+        // Different versions in assets and package spec, greater in spec
+        [Fact]
+        public async Task GetInstalledVersion_WithAssetsVersionLower_Error()
```

Comment:
This test seems to be on testing restore, it never runs GetInstalledVersion does it? 

Did you mean toa dd something else?